openstf/model/objective.py

```diff
@@ -55,6 +121,11 @@ def __call__(self, trial: optuna.trial.FrozenTrial) -> float:
         Returns:
             float: Mean absolute error for this trial.
         """
+        # Check elapsed time and after n minutes end trial and stop the study
+        elapsed = datetime.utcnow() - self.start_time
```

Comment:
Why is this custom stopping criteria needed? Does optuna not have all kind of functionality to determine when to stop?

Reply:
The custom stopping criteria allows us to use trial.study.stop() this will first finish the trial and after the trial finished it will stop the study. Using the time limit from optuna it will stop the study during a trial.

Reply:
How do you know this: "Using the time limit from optuna it will stop the study during a trial". I would think `optune.study.Study.optimize` would check total duration after a trial?